social-messenger-ts/src/App.tsx

```diff
@@ -143,14 +144,13 @@ const App = () => {
 
   return (
     <Chat client={chatClient} theme={`messaging ${theme}`}>
-      <div id='mobile-channel-list' onClick={toggleMobile}>
+      <div className='messaging__sidebar' id='mobile-channel-list' onClick={toggleMobile}>
+        <MessagingChannelListHeader onCreateChannel={() => setIsCreating(!isCreating)} theme={theme} />
         <ChannelList
           filters={filters}
           sort={sort}
           options={options}
-          List={(props) => (
-            <MessagingChannelList {...props} onCreateChannel={() => setIsCreating(!isCreating)} />
-          )}
+          List={(props) => <MessagingChannelList {...props} />}
```

Comment:
This line, while looking innocent, is a performance hit. It creates a new fresh component instance with each render cycle, causing the MessagingChannelList to mount/unmount. Now that you have cleared the creating part, you can reduce it to

```suggestion
          List={MessagingChannelList}
```